embedded-graphics/src/primitives/circle.rs

```diff
@@ -132,26 +134,21 @@ where
 {
     type Item = Pixel<C>;
 
-    // https://stackoverflow.com/questions/1201200/fast-algorithm-for-drawing-filled-circles
+    // https://stackoverflow.com/a/1237519/383609
+    // https://stackoverflow.com/questions/1201200/fast-algorithm-for-drawing-filled-circles#comment80182898_1237519
     fn next(&mut self) -> Option<Self::Item> {
         // If border or stroke colour is `None`, treat entire object as transparent and exit early
         if self.style.stroke_color.is_none() && self.style.fill_color.is_none() {
             return None;
         }
 
-        let radius = self.radius as i32 - self.style.stroke_width_i32() + 1;
-        let outer_radius = self.radius as i32;
-
-        let radius_sq = radius * radius;
-        let outer_radius_sq = outer_radius * outer_radius;
-
         loop {
             let t = self.p;
-            let len = t.x * t.x + t.y * t.y;
+            let len = t.x.pow(2) + t.y.pow(2);
 
-            let is_border = len > radius_sq - radius && len < outer_radius_sq + radius;
+            let is_border = len > self.inner_threshold && len < self.outer_threshold;
 
-            let is_fill = len <= outer_radius_sq + 1;
+            let is_fill = len < self.outer_threshold;
```

Comment:
Should these use `<` or `<=` for the comparisons? The size 4 circle in the debug example looks better with `<=` IMO (and matches an equally sized circle drawn in GIMP). But this change breaks size 1 rendering.

Reply:
Hm yeah it does look a lot better. I've made them `<=` in eab8abd.

I also added a kludge in d6b0b86 that fixes tiny circles. I can't really think of a better way of doing this at the moment so please let me know. It does _work_, but it doesn't feel clean.